platform/location.hpp

```diff
@@ -17,7 +17,9 @@ namespace location
     ENoError = 0,
     ENotSupported,
     EDenied,
-    EGPSIsOff
+    EGPSIsOff,
+    ETimeout,
```

Comment:
1. There likely was a reason why timeout "error" was not added initially. Did you check the existing code?
2. MWMLocationStatus should be updated if this enum is updated. But let's decide first if it is really necessary.
3. LocationState.java also should be updated. ditto.

Reply:
> There likely was a reason why timeout "error" was not added initially. Did you check the existing code?

In the Qt positioning [timeout is definitely an error](https://doc.qt.io/qt-6/qgeopositioninfosource.html#Error-enum)
And I have checked the existing code, have not found anything, have I missed something?

Reply:
@biodranik Can I proceed to the `2.` and `3.` points?

Reply:
Yes.

From the Qt docs it looks like timeout error can be recoverable and unrecoverable. Looks like some additional logic is needed to properly handle different cases.

It would be better to add comments with a link and text that this is used only in Qt now.

Reply:
Done. PTAL

Reply:
@biodranik For your kind review